client/ayon_maya/plugins/inventory/connect_ornatrix_rig.py

```diff
@@ -0,0 +1,178 @@
+        # Categorize containers by product type.
+        containers_by_product_type = defaultdict(list)
+        repre_ids = {
+            container["representation"]
+            for container in containers
+        }
+        repre_contexts_by_id = get_repres_contexts(repre_ids)
+        for container in containers:
+            repre_id = container["representation"]
+            repre_context = repre_contexts_by_id[repre_id]
+
+            product_type = repre_context["product"]["productType"]
+            containers_by_product_type[product_type].append(container)
+
+        # Validate to only 1 source container.
+        source_containers = containers_by_product_type.get("animation", [])
+        source_containers += containers_by_product_type.get("pointcache", [])
+        source_container_namespaces = [
+            x["namespace"] for x in source_containers
+        ]
+        message = (
+            "{} animation containers selected:\n\n{}\n\nOnly select 1 of type "
+            "\"animation\" or \"pointcache\".".format(
+                len(source_containers), source_container_namespaces
+            )
+        )
+        if len(source_containers) != 1:
+            self.display_warning(message)
+            return
+
+        source_container = source_containers[0]
+        source_repre_id = source_container["representation"]
+        source_namespace = source_container["namespace"]
+
+        # Validate source representation is an alembic.
+        source_path = get_representation_path(
+            repre_contexts_by_id[source_repre_id]["representation"]
+        ).replace("\\", "/")
+        message = "Animation container \"{}\" is not an alembic:\n{}".format(
+            source_container["namespace"], source_path
+        )
+        if not source_path.endswith(".abc"):
+            self.display_warning(message)
+            return
+
+        ox_rig_containers = containers_by_product_type.get("oxrig")
+        if not ox_rig_containers:
+            self.display_warning(
+                "Select at least one oxrig container"
+            )
+            return
+
+        for container in ox_rig_containers:
+            rig_namespace = container["namespace"]
+            repre_id = container["representation"]
+            maya_file = get_representation_path(
+                repre_contexts_by_id[repre_id]["representation"]
+            )
+
+            # Get base filename without extension
+            # TODO: We should actually get the actual representation paths
+            #   through the parent version entity so that we get the
+            #   representation's paths supporting the template system instead
+            #   of assuming the files live next to the loaded rig file directly
+            # of the `.oxg.zip` and the `.rigsettings` instead of computing the
+            # relative paths
+            if maya_file.endswith(".oxg.zip"):
+                base = maya_file[-len(".oxg.zip")]  # strip off multi-dot ext
```

Comment:
```suggestion
               base = [x.strip() for x in maya_file.split('.')][0]  # strip off multi-dot ext
```
I hit weird bug when using oxg.zip for connecting ox rig, where the base variable goes empty the path of rigsettings file is `.rigsettings`

Reply:
Thanks will look into it.

Reply:
Should be fixed with https://github.com/ynput/ayon-maya/pull/1/commits/01de96ce3561d0880f0acedd9e65a436374317ce